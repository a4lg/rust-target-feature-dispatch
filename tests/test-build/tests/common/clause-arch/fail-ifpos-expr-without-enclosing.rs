fn main() {
    let _: i32 = target_feature_dispatch::target_feature_dispatch! {
        // SPECIFIC: ARCH_COMPILER_VARIANT
        if "aarch64" || "arm" || "arm64ec" || "avr" || "bpf" || "csky" || "hexagon" || "loongarch64" || "m68k" || "mips" || "mips32r6" || "mips64" || "mips64r6" || "msp430" || "nvptx64" || "powerpc" || "powerpc64" || "riscv32" || "riscv64" || "s390x" || "sparc" || "sparc64" || "wasm32" || "wasm64" || "x86" || "x86_64" || "xtensa" {
            // ERROR: an expression must be enclosed to distinguish between
            //        this macro's special syntax and the regular expression.
            1
        } else {
            compile_error!(concat!(
                "This position should not be expanded ",
                "(maybe Rust compiler added an architecture?)"
            ));
        }
    };
}
