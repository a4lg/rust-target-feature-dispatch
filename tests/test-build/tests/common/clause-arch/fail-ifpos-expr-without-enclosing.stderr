error: unsupported or invalid architecture clause
  --> tests/common/clause-arch/fail-ifpos-expr-without-enclosing.rs:2:18
   |
2  |       let _: i32 = target_feature_dispatch::target_feature_dispatch! {
   |  __________________^
3  | |         // SPECIFIC: ARCH_COMPILER_VARIANT
4  | |         if "aarch64" || "arm" || "arm64ec" || "avr" || "bpf" || "csky" || "hexagon" || "loongarch64" || "m68k" || "mips" || "mips32r6" ||...
...  |
14 | |     };
   | |_____^
   |
   = note: this error originates in the macro `$crate::target_feature_dispatch` which comes from the expansion of the macro `target_feature_dispatch::target_feature_dispatch` (in Nightly builds, run with -Z macro-backtrace for more info)
